crates/storage/derive/src/lib.rs

```diff
@@ -15,163 +15,94 @@
 //! Custom derive for `ink_storage` traits.
 //!
 //! This crate provides helpers to define your very own custom storage data
-//! structures that work along the `ink_storage` data structures by implementing
-//! `SpreadLayout` and `PackedLayout` traits.
-//!
-//! See [Spread vs. Packed](https://ink.substrate.io/datastructures/spread-packed-layout)
-//! for more details of these two root strategies.
-//!
-//! # Examples
-//!
-//! ```no_run
-//! use ink_storage::traits::{SpreadLayout, push_spread_root};
-//! use ink_primitives::Key;
-//!
-//! # ink_env::test::run_test::<ink_env::DefaultEnvironment, _>(|_| {
-//! // Enum
-//! #[derive(SpreadLayout)]
-//! enum Vote {
-//!     Yes,
-//!     No
-//! }
-//!
-//! // Strucut
-//! #[derive(SpreadLayout)]
-//! struct NamedFields {
-//!     a: u32,
-//!     b: [u32; 32],
-//! };
-//!
-//! // Created a custom structure and told which key to update.
-//! push_spread_root(&NamedFields{ a: 123, b: [22; 32] }, &mut Key::from([0x42; 32]));
-//! # Ok(())
-//! # });
-//! ```
+//! structures that work along the `ink_storage` data structures.
 
 extern crate proc_macro;
 
-mod packed_layout;
-mod spread_allocate;
-mod spread_layout;
+mod item;
+mod key_holder;
 mod storage_layout;
 
 #[cfg(test)]
 mod tests;
 
 use self::{
-    packed_layout::packed_layout_derive,
-    spread_allocate::spread_allocate_derive,
-    spread_layout::spread_layout_derive,
+    item::item_derive,
+    key_holder::key_holder_derive,
     storage_layout::storage_layout_derive,
 };
 synstructure::decl_derive!(
-    [SpreadLayout] =>
-    /// Derives `ink_storage`'s `SpreadLayout` trait for the given `struct` or `enum`.
+    [Item] =>
+    /// Derives `ink_storage`'s `Item` trait for the given `struct` or `enum`.
     ///
     /// # Examples
     ///
     /// ```
-    /// use ink_primitives::Key;
     /// use ink_storage::traits::{
-    ///     SpreadLayout,
-    ///     push_spread_root,
-    ///     pull_spread_root,
-    ///};
+    ///     Item,
+    ///     StorageKey,
+    ///     AutoItem,
+    ///     AutoKey,
+    ///     ManualKey,
+    /// };
+    /// use ink_primitives::traits::Storable;
     ///
-    /// # ink_env::test::run_test::<ink_env::DefaultEnvironment, _>(|_| {
-    /// #[derive(SpreadLayout)]
+    /// #[derive(Default, Item, Storable)]
```

Comment:
It is because of the blanket implementation of `Storable`. Rust tries to describe what should be implemented to use blanket implementation=) We can do nothing with it